amplitude.py

```diff
@@ -20,18 +23,40 @@
 # specifying wbits=0 but that didn't work for me locally. This did.
 ZLIB_HEADER_SIZE = 32
 
-def handle (message):
+def kms_decrypt_env(key):
+    """Decrypt environment variable"""
+    return kms_decrypt(b64decode(os.environ[key]))
+
+def kms_decrypt(encrypted_data):
+    """Decrypt KMS variables"""
+    res = boto3.client("kms").decrypt(
+        CiphertextBlob=encrypted_data,
+    )
+    return res["Plaintext"].decode("utf-8")
+
+if os.environ["LAMBDA_TASK_ROOT"]:
+    AMPLITUDE_API_KEY = str(kms_decrypt_env("FXA_AMPLITUDE_API_KEY"))
+    HMAC_KEY = str(kms_decrypt_env("FXA_AMPLITUDE_HMAC_KEY"))
+
+def handle (message, context):
     # http://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html
-    records = json.loads(message)["Records"]
+    if (type(message) is str):
+        message = json.loads(message)
+    records = message["Records"]
+
     for record in records:
         if record["eventSource"] != "aws:s3":
             continue
 
+        print record["s3"]["bucket"]["name"], record["s3"]["object"]["key"]
         s3 = boto3.resource("s3", region_name=record["awsRegion"])
         s3_object = s3.Object(record["s3"]["bucket"]["name"], record["s3"]["object"]["key"])
+        tmpfileobj, tmppath = tempfile.mkstemp()
+        s3_object.download_file(tmppath)
```

Comment:
Did you do this because the streaming approach doesn't work? If that's the case it's probably worth us spending a little bit of time trying to make it work.

And if we can't make it work, we should make this change as a separate commit so it's visible in the history.